R/feature.align.R

```diff
@@ -62,119 +120,107 @@ feature.align <- function(features,
         }
         
         # find relative retention time tolerance level
-        all.ft <- find.tol.time(mz_values,
-                                chr,
-                                lab,
-                                number_of_samples = number_of_samples,
-                                mz_tol_relative = mz_tol_relative,
-                                rt_tol_relative = rt_tol_relative,
-                                mz_tol_absolute = mz_tol_absolute,
-                                do.plot = do.plot)
-        rt_tol_relative <- all.ft$chr.tol
+        # also does some preprocessing grouping steps
+        all_features <- find.tol.time(mz_values,
+                                      rt,
+                                      sample_id,
+                                      number_of_samples = number_of_samples,
+                                      mz_tol_relative = mz_tol_relative,
+                                      rt_tol_relative = rt_tol_relative,
+                                      mz_tol_absolute = mz_tol_absolute,
+                                      do.plot = do.plot)
+        rt_tol_relative <- all_features$rt.tol
         
         message("**** performing feature alignment ****")
         message(paste("m/z tolerance level: ", mz_tol_relative))
         message(paste("time tolerance level:", rt_tol_relative))
         
-        aligned.ftrs <- pk.times <- rep(0, 4 + number_of_samples)
-        mz.sd.rec <- 0
-        
-        labels <- unique(all.ft$grps)
+        # create zero vectors of length number_of_samples + 4 ?
+        aligned_features <- pk.times <- rep(0, 4 + number_of_samples)
+        mz_sd <- 0
         
+        labels <- unique(all_features$grps)
         area <- grps <- mz_values
         
         # grouping the features based on their m/z values (assuming the tolerance level)
         sizes <- c(0, cumsum(sapply(features, nrow)))
         for (i in 1:number_of_samples) {
-            this <- features[[i]]
-            sel <- which(all.ft$lab == i)
-            that <- cbind(all.ft$mz[sel], all.ft$chr[sel], all.ft$grps[sel])
-            this <- this[order(this[, 1], this[, 2]),]
-            that <- that[order(that[, 1], that[, 2]),]
+            sample <- features[[i]]
+            # order by m/z then by rt
+            sample <- sample[order(sample[, 1], sample[, 2]),]
+            
+            # select preprocessed features belonging to current sample
+            group_ids <- which(all_features$lab == i)
+            # select m/z, rt and their group ID
+            sample_grouped <- cbind(all_features$mz[group_ids], all_features$rt[group_ids], all_features$grps[group_ids])
+            # order them again? should be ordered already...
+            sample_grouped <- sample_grouped[order(sample_grouped[, 1], sample_grouped[, 2]),]
```

Comment:
Did you try to run tests without this line?

Reply:
Good point, without that the test case fails.